libs/experimental/langchain_experimental/llms/ollama_functions.py

```diff
@@ -354,6 +357,86 @@ def _generate(
             generations=[ChatGeneration(message=response_message_with_functions)]
         )
 
+    async def _agenerate(
```

Comment:
Isn't this duplicating the implementation of the sync method? I thought BaseModel already takes care of this automatically under the hood?